Add metrics and refactor tracing

asteurer opened the pull request:
This adds metrics to the WIT as well as some changes to the tracing portion.

calebschoepp commented on README.md:
```diff
@@ -9,6 +9,7 @@ Phase 0
 ### Champions
 
 - [Caleb Schoepp](https://github.com/calebschoepp)
+- [Andrew Steurer](https://github.com/asteurer)
```
I'm totally fine with you co-championing this and marking it down to make it official. But, let's do it in a separate PR to avoid appearances of trying to sneak this in.

Also note that the more canonical and important place to mark this down would be opening a PR for [here](https://github.com/WebAssembly/WASI/blob/main/Proposals.md#phase-0---pre-proposal-cg).

asteurer replied:
Happy to separate that into another PR. Is there any nuance to being a champion that I should be aware of?

calebschoepp replied:
Probably 🤷, but nothing that I'm specifically aware of to warn you.

asteurer committed:
feat(metrics): refactoring tracing and adding metrics

Signed-off-by: Andrew Steurer <94206073+asteurer@users.noreply.github.com>

fix: making changes for CI

Signed-off-by: Andrew Steurer <94206073+asteurer@users.noreply.github.com>

fix: trying some experimenting with CI

Signed-off-by: Andrew Steurer <94206073+asteurer@users.noreply.github.com>

fix: more CI experiments

Signed-off-by: Andrew Steurer <94206073+asteurer@users.noreply.github.com>

fix: CI removing extra things

Signed-off-by: Andrew Steurer <94206073+asteurer@users.noreply.github.com>
